src/Plugins/LevelDBStore/IO/Data/LevelDB/Native.cs

```diff
@@ -83,182 +91,169 @@ public static class Native
 
         [DllImport("libleveldb", CharSet = CharSet.Ansi, CallingConvention = CallingConvention.Cdecl)]
-        public static extern void leveldb_iter_get_error(IntPtr /*Iterator*/ iterator, out IntPtr error);
+        public static extern void leveldb_iter_get_error(nint /*Iterator*/ iterator, out nint error);
         #endregion
 
         #region Options
         [DllImport("libleveldb", CharSet = CharSet.Ansi, CallingConvention = CallingConvention.Cdecl)]
-        public static extern IntPtr leveldb_options_create();
+        public static extern nint leveldb_options_create();
 
         [DllImport("libleveldb", CharSet = CharSet.Ansi, CallingConvention = CallingConvention.Cdecl)]
-        public static extern void leveldb_options_destroy(IntPtr /*Options*/ options);
+        public static extern void leveldb_options_destroy(nint /*Options*/ options);
 
         [DllImport("libleveldb", CharSet = CharSet.Ansi, CallingConvention = CallingConvention.Cdecl)]
-        public static extern void leveldb_options_set_create_if_missing(IntPtr /*Options*/ options, [MarshalAs(UnmanagedType.U1)] bool o);
+        public static extern void leveldb_options_set_create_if_missing(nint /*Options*/ options, bool o);
 
         [DllImport("libleveldb", CharSet = CharSet.Ansi, CallingConvention = CallingConvention.Cdecl)]
-        public static extern void leveldb_options_set_error_if_exists(IntPtr /*Options*/ options, [MarshalAs(UnmanagedType.U1)] bool o);
+        public static extern void leveldb_options_set_error_if_exists(nint /*Options*/ options, bool o);
 
         [DllImport("libleveldb", CharSet = CharSet.Ansi, CallingConvention = CallingConvention.Cdecl)]
-        public static extern void leveldb_options_set_info_log(IntPtr /*Options*/ options, IntPtr /* Logger */ logger);
+        public static extern void leveldb_options_set_info_log(nint /*Options*/ options, nint /* Logger */ logger);
 
         [DllImport("libleveldb", CharSet = CharSet.Ansi, CallingConvention = CallingConvention.Cdecl)]
-        public static extern void leveldb_options_set_paranoid_checks(IntPtr /*Options*/ options, [MarshalAs(UnmanagedType.U1)] bool o);
+        public static extern void leveldb_options_set_paranoid_checks(nint /*Options*/ options, bool o);
 
         [DllImport("libleveldb", CharSet = CharSet.Ansi, CallingConvention = CallingConvention.Cdecl)]
-        public static extern void leveldb_options_set_env(IntPtr /*Options*/ options, IntPtr /*Env*/ env);
+        public static extern void leveldb_options_set_env(nint /*Options*/ options, nint /*Env*/ env);
 
         [DllImport("libleveldb", CharSet = CharSet.Ansi, CallingConvention = CallingConvention.Cdecl)]
-        public static extern void leveldb_options_set_write_buffer_size(IntPtr /*Options*/ options, UIntPtr size);
+        public static extern void leveldb_options_set_write_buffer_size(nint /*Options*/ options, long size);
```

Comment:
It's always long? IntPtr could be int in different machines

Reply:
https://github.com/google/leveldb/blob/23e35d792b9154f922b8b575b12596a4d8664c65/include/leveldb/c.h#L182-L183

No, is don't, `size_t` it's platform dependant, we need to use `UIntPtr`

Reply:
Setting it as `long`  will make up for it. memory will already be allocated 64 or 32-bit because as long it's bigger than int.